src/stytch-provider.ts

```diff
@@ -44,18 +50,150 @@ function StytchProvider(this: any, options: StytchProviderOptions) {
-              let list = res.results.map((data: any) => entize(data, {
+	      q = Object.keys(q).reduce((acc: any, key: any) => {
+		if(key.endsWith('$')) {
+		  key = key.replace(/.$/, '')
+		  acc[key] = q[key+'$']
+		} else acc[key] = q[key]
+		return acc
+	      }, {})
+	      let res: any = null
+
+	      try {
+                res = await this.shared.sdk.users.search(q)
+	      }catch(err_res) {
+		res = err_res
+	      }
+	      check_status.call(this, res)
+
+	      let list = res.results.map((data: any) => entize(data, {
                 field: {
                   id: { src: 'user_id' }
                 }
-              }))
+	      })
+
+	      )
               return list
             }
           },
+
+	  load: {
+	    action: async function(this: any, entize: any, msg: any) {
+	      let id = msg?.q?.id
+	      let res: any = null
+
+	      id == null ? this.fail('invalid id') : null
+
+
+	      try {
+                res = await this.shared.sdk.users.get(id)
+	      }catch(err_res) {
+		res = err_res
+	      }
+	      check_status.call(this, res)
+
+	      // TODO: something like: (entize({'res': res} as ProviderRes)) for more structure?
+	      return entize(res, {
+                field: {
+                  id: { src: 'user_id' }
+                }
+              })
+
+	    }
+	  },
+	  
+          save: {
+	    action: async function(this: any, entize: any, msg: any) {
+	      let ent: any = msg.ent
+	      let id = ent.id
+	      let user: any = ent['user'] || {}
+	      let params: Array<any> = []
+	      let api_call: string
+
+	      let res: any = null
+
+	      api_call = id == null ?
+	        (params = [ user ], 'create') : (params = [ id, user ], 'update')
+	      // invalid body parameters
+	      if(0 === Object.keys(user).length) {
+	        this.fail('empty body parameters', 'user field')
+	      }
+
+	      try {
+                res = await this.shared.sdk.users[api_call](...params)
+	      }catch(err_res) {
+		res = err_res
+	      }
+	      check_status.call(this, res)
+              
+	      return entize(res, {
+                field: {
+                  id: { src: 'user_id' }
+                }
+              })
+
+	    }
+	  },
+
+	  remove: {
+            action: async function(this: any, entize: any, msg: any) {
+	      let id = msg?.q?.id
+	      let res: any = null
+
+	      id == null ? this.fail('invalid id') : null
+	      try {
+                res = await this.shared.sdk.users.delete(id)
+	      }catch(err_res) {
+		res = err_res
+	      }
+	      check_status.call(this, res)
+
+	      return entize(res, {
+                field: {
+                  id: { src: 'user_id' }
+                }
+              })
+
+	    }
+
+	  },
+
+        }
+      },
+      
+      session: {
+        cmd: {
+          list: {
+            action: async function(this: any, entize: any, msg: any) {
+              let q = msg.q || {}
+	      
+              q = Object.keys(q).reduce((acc: any, key: any) => {
```

Comment:
while this should not in any case be here, if it was to be, this cut and paste of intricate code is not DRY